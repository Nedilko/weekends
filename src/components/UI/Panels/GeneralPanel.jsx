import PropTypes from 'prop-types'

GeneralPanel.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
}

GeneralPanel.defaultProps = {
  className: '',
}

function GeneralPanel({ children, className }) {
  return (
    <div
      className={`absolute overflow-auto rounded-md bg-white shadow-md dark:bg-zinc-900 ${className}`}
      onClick={(e) => e.stopPropagation()}
    >
      {children}
    </div>
  )
}

export default GeneralPanel
