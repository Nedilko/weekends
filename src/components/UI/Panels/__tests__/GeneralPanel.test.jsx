import GeneralPanel from '@UI/Panels/GeneralPanel'
import { render } from '@utils/test-utils'

describe('panel', () => {
  it('renders text', () => {
    const { container, getByText } = render(
      <GeneralPanel>some text</GeneralPanel>
    )
    expect(getByText('some text')).toBeInTheDocument()
    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
