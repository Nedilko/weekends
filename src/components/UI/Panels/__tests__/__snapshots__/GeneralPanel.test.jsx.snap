// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralPanel should render content 1`] = `
<div>
  <div
    class="absolute overflow-auto rounded-md bg-white shadow-md dark:bg-zinc-900 "
  >
    some content
  </div>
</div>
`;
